Firefly/Views/Transactions/TransactionEdit.swift

```diff
@@ -0,0 +1,291 @@
+                                    postTransactionData.transactions?[index].date =
+                                        dateFormatter.string(from: newValue)
+                                }
+                            )
+                        ).datePickerStyle(.compact)
+
+                        TextField(
+                            "Notes",
+                            text: Binding(
+                                get: {
+                                    transaction.notes ?? ""
+                                },
+                                set: { newValue in
+                                    postTransactionData.transactions?[index].notes = newValue
+                                }), axis: .vertical
+                        ).lineLimit(5)
+
+                        //TODO: add category button
+                        //TODO: budget select
+
+                    } header: {
+                        HStack {
+                            Text("Details")
+                            Spacer()
+                            if isSplitTransaction(postTransactionData) {
+                                Text("Split #\(index + 1)")
+                            }
+                        }
+                    }
+
+                }
+
+            }
+            .safeAreaInset(
+                edge: .bottom,
+                content: {
+                    MasterButton(
+                        icon: "square.and.pencil", label: "Edit Transaction", fullWidth: true,
+                        disabled: submitLoading,
+                        action: {
+                            Task {
+                                do {
+                                    try await submitTransaction()
+                                } catch {
+                                    print("ERROR EDITING: \(error)")
+                                }
+                            }
+                        }
+                    ).padding(.horizontal, 16).padding(.bottom, 8)
+                }
+            )
+            .navigationTitle("Edit Transaction")
+            .navigationBarTitleDisplayMode(.inline)
+            .navigationBarItems(trailing: Button("Cancel") { dismiss() })
+            .toast(isPresenting: $showToast, tapToDismiss: false) {
+                toastParams
+            }
+        }
+
+        .onAppear {
+            loadAutocomplete()
+        }
+    }
+
+    func loadAutocomplete() {
+        Task {
+            do {
+                //Categories
+                categories = try await createCategoriesAutocompleteArray()
+                //Transactions (for Description Autocomplete)
+                descriptions = try await fetchTransactionAutocomplete()
+                //Accounts
+                accounts = try await fetchAccountsAutocomplete()
+
+            } catch {
+                print("Error: \(error)")
+            }
+        }
+    }
+
+    func submitTransaction() async throws {
```

Comment:
**suggestion:** Improve error handling in submitTransaction

Consider implementing a more robust error handling approach in `submitTransaction`. Ensure that the loading state is always reset, regardless of whether the operation succeeds or fails. This could be achieved using a defer statement or a do-catch-finally block.